crates/subgraph/src/orderbook_client.rs

```diff
@@ -353,4 +353,31 @@ impl OrderbookSubgraphClient {
         }
         Ok(all_pages_merged)
     }
+
+    /// Fetch all trades from all orders within a time period
+    pub async fn all_trades_list(
+        &self,
+        start_timestamp: Option<u64>,
+        end_timestamp: Option<u64>,
+    ) -> Result<Vec<Trade>, OrderbookSubgraphClientError> {
+        let orders = self.orders_list_all().await?;
+
+        let mut all_trades = Vec::new();
+        for order in orders {
```

Comment:
I don't like the performance output of this, but to fix that, would had been out of the scope of the PR.